GCC 7.1 build error in CString.h

GCC 7.1 errors with the following when building. Works fine on GCC 5.x.

```c++
[  2%] Building CXX object CMakeFiles/MfgToolLib.dir/CmdOperation.cpp.o
/usr/bin/c++  -DMfgToolLib_EXPORTS -I/home/thrackerzod/src/3rdparty/NXPmicro/mfgtools/MfgToolLib -I/usr/include/libusb-1.0  -std=c++11 -O0 -fpic -I/usr/lib/jvm/java-8-oracle/include -I/usr/lib/jvm/java-8-oracle/include/linux -fpermissive -g -Wno-write-strings -fPIC   -o CMakeFiles/MfgToolLib.dir/CmdOperation.cpp.o -c /home/thrackerzod/src/3rdparty/NXPmicro/mfgtools/MfgToolLib/CmdOperation.cpp
In file included from /home/thrackerzod/src/3rdparty/NXPmicro/mfgtools/MfgToolLib/lnx_def.h:46:0,
                 from /home/thrackerzod/src/3rdparty/NXPmicro/mfgtools/MfgToolLib/stdafx.h:83,
                 from /home/thrackerzod/src/3rdparty/NXPmicro/mfgtools/MfgToolLib/CmdOperation.cpp:32:
/home/thrackerzod/src/3rdparty/NXPmicro/mfgtools/MfgToolLib/CString.h: In member function ‘void CTString<T>::TrimLeft(T)’:
/home/thrackerzod/src/3rdparty/NXPmicro/mfgtools/MfgToolLib/CString.h:179:33: error: no matching function for call to ‘CTString<T>::at(std::__cxx11::basic_string<char>::iterator)’
    while (this->at(this->begin()) == chr){

[...]
/home/thrackerzod/src/3rdparty/NXPmicro/mfgtools/MfgToolLib/CString.h:204:32: error: no matching function for call to ‘CTString<T>::at(std::__cxx11::basic_string<char>::iterator)’
      if (this->at(this->begin()) == chr[i]){

[...]
/home/thrackerzod/src/3rdparty/NXPmicro/mfgtools/MfgToolLib/CString.h:223:30: error: no matching function for call to ‘CTString<T>::at(std::__cxx11::basic_string<char>::iterator)’
      if (this->at(this->end()) == chr[i]){
```

The errors all seem to be caused by trying to use an iterator as an index.

--- a/MfgToolLib/CString.h
+++ b/MfgToolLib/CString.h
@@ -176,7 +176,7 @@ public: CTString() : tstring() { }
 
 
 		void TrimLeft(T  chr){
-			while (this->at(this->begin()) == chr){
+			while (this->at(0) == chr){
 				this->erase(this->begin());
 				if (this->empty())
 					return;
@@ -201,7 +201,7 @@ public: CTString() : tstring() { }
 			while (present){
 				present = false;
 				for (i = 0; i < _tcslen(chr); i++){
-					if (this->at(this->begin()) == chr[i]){
+					if (this->at(0) == chr[i]){
 						present = true;
 						this->erase(this->begin());
 						break;
@@ -220,7 +220,7 @@ public: CTString() : tstring() { }
 			while (present){
 				present = false;
 				for (i = 0; i < _tcslen(chr); i++){
-					if (this->at(this->end()) == chr[i]){
+					if (this->at(this->length()-1) == chr[i]){
 						present = true;
 						this->erase(this->end());
 						break;

